app/api-v1/routes/attachment/{id}.js

```diff
@@ -1,8 +1,26 @@
 // eslint-disable-next-line no-unused-vars
+const { BadRequestError } = require('../../../utils/errors')
 module.exports = function (attachmentService) {
   const doc = {
     GET: async function (req, res) {
-      res.status(500).json({ message: 'Not Implemented' })
+      const response = await attachmentService.getAttachmentByID(req.params.id)
+
+      if (req.headers.accept.includes('json') && response.filename === 'json') {
```

Comment:
I think we need to do a more correct parsing of the accept header here. I'm not sure if there's an existing module (hopefully) but the standard is well documented https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept. Essentially if the file is JSON and the accept header prefers JSON we should return that. If the file is not json but there is an accept entry for `application/octet-stream`, `application/*` or `*/*` we should return the octet stream. We might need to work out some test cases to enumerate the different behaviours we want then use that to test our implmentation

Reply:
At the moment anything that isn't `application/json` will return an `octect-stream` by default.

Reply:
At the moment it defaults to an `octet-stream` so do you want individual accept type checking?